### ISSUE TITLE
warning when update react lifecycle from componentWillReceiveProps to UNSAFE_componentWillReceiveProps

### ISSUE THREAD

**Issue by slim-hmidi** [eslint/eslint#10430]
**Please describe what the rule should do:**
Due to the update of [react v16.3.0](https://reactjs.org/blog/2018/03/29/react-v-16-3.html), there are some update on the name of lifcycles such as `componentWillReceiveProps` will be `UNSAFE_componentWillReceiveProps`. I got this error : 
```
UNSAFE_componentWillReceiveProps is not in camel case.
```
I fix this by disabling eslint line.
**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ ] Warns about a potential error
[X] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
UNSAFE_componentWillReceiveProps(nextProps) {
  /...
  }
```
**.eslintrc.json:** 
```
{
  "env": {
    "browser": true
  },
  "rules": {
    "react/forbid-prop-types": "off",
    "jsx-a11y/anchor-is-valid": "off"
  },
  "parser": "babel-eslint",
  "settings": {
    "import/resolver": {
      "webpack": {
          "config": "webpack.common.js"
        }
    }
  }
}
```

**eslint:** 4.17.0

**Comment by platinumazure** [comment]
I think you could work around this by migrating from the `camelcase` rule to `id-match`. `id-match` can be configured with a regular expression; you can allow the `UNSAFE_` prefix, but otherwise require camelcase.

**Comment by slim-hmidi** [comment]
@platinumazure I think `id-match` should take into consideration many form of strings and it will be painful. I should concatenate different regexp to match different strings.

**Comment by kaicataldo** [comment]
This is a dupe of https://github.com/eslint/eslint/issues/10503, which has a proposed solution to the problem. Closing in favor of that issue - feel free to comment there.
